src/backend/src/services/tasks.services.ts

```diff
@@ -51,7 +55,18 @@ export default class TasksService {
     const isProjectLeadOrManager =
       createdBy.userId === requestedWbsElement.projectLeadId || createdBy.userId === requestedWbsElement.projectManagerId;
 
-    if (!isLeadership(createdBy.role) && !isProjectLeadOrManager && !teams.some((team) => isUserOnTeam(team, createdBy))) {
+    const curWorkPackages = requestedWbsElement.project?.workPackages;
+
+    const isWorkPackageLead = curWorkPackages?.some((workPackage) => {
```

Comment:
After we resolve the comment on `curWorkPackages `variable, we can replace this `curWorkPackages?` with `curWorkPackages `since we know that `curWorkPackages `is defined.